app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawerTest.kt

```diff
@@ -0,0 +1,227 @@
+        val startTime2 = startTime1 + duration1 + 30 // 00:15am, next day
+
+        val lecture1 = createLecture(room = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(room = 0, date = conferenceDate, time = startTime2)
+        val lectures = listOf(lecture1, lecture2)
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lecture1Params = layoutParams[lecture1]
+        val secondLectureParams = layoutParams[lecture2]
+        val gapMinutes = 30
+
+        assertMargins(lecture1Params, 0, gapMinutes)
+        assertMargins(secondLectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams overlapping lecture in same room - should cut first lecture duration to match next lecture start`() {
+        val duration1 = 45
+        val startTime1 = 10 * 60 // 10:00am
+        val startTime2 = startTime1 + duration1 - 10 // 10:35am (10 minutes overlap)
+
+        val lecture1 = createLecture(room = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(room = 0, date = conferenceDate, time = startTime2)
+        val lectures = listOf(lecture1, lecture2)
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParams = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val lecture1Params = layoutParams[lecture1]
+        val secondLectureParams = layoutParams[lecture2]
+
+        assertMargins(lecture1Params, 0, 0)
+        assertMargins(secondLectureParams, 0, 0)
+    }
+
+    @Test
+    fun `calculateLayoutParams overlapping lecture in another room - should not cut any lecture`() {
+        val duration1 = 45
+        val startTime1 = 10 * 60 // 10:00am
+        val startTime2 = startTime1 + duration1 - 10 // 10:35am (10 minutes overlap)
+
+        val lecture1 = createLecture(room = 0, date = conferenceDate, time = startTime1, _duration = duration1)
+        val lecture2 = createLecture(room = 1, date = conferenceDate, time = startTime2)
+        val lectures = listOf(lecture1, lecture2)
+        conference.calculateTimeFrame(lectures) { dateUTC: Long -> Moment(dateUTC).minuteOfDay }
+
+        val layoutParamsRoom1 = LectureViewDrawer.calculateLayoutParams(0, lectures, standardHeight, conference)
+        val layoutParamsRoom2 = LectureViewDrawer.calculateLayoutParams(1, lectures, standardHeight, conference)
+        val lecture1Params = layoutParamsRoom1[lecture1]
+        val secondLectureParams = layoutParamsRoom2[lecture2]
+
+        assertMargins(lecture1Params, 0, 0)
+        assertMargins(secondLectureParams, 35, 0)
+    }
+
+    private fun assertMargins(secondLectureParams: LinearLayout.LayoutParams?, top: Int, bottom: Int) {
```

Comment:
Rename `secondLectureParams` to `lectureParams` because it is not always the 2nd.